### ISSUE TITLE
Version 3.11.0: Auto-tagging of meetings in new version causes problems

### ISSUE THREAD

**Issue by OSAwebsite** [code4recovery/12-step-meeting-list#405]
I am entering this as a bug but I realise it might not be considered that by the code-writers.

In short:
it seems the newest TSML version auto-tags meetings and this is causing random errors.
I would like to know what is triggering the tagging and with that I mean: is a meeting online or in person or both?

I have a lot of meetings, after updating on my staging site, that show 'in person' and there is nothing in the meeting information that justifies that.

I would like to get more clarity about what triggers the tagging.

Ideally I would like all coding to use the check box options to draw conclusions and not do so in a way that is not transparent for me as the admin of the website.

I don't know the coding, but I remember there was a change during Covid at some point, where we no longer had to indicate whether a meeting was online or not. The coding, I was told, just checked if there was a valid URL for Zoom (and similar online platforms) to come to the conclusion that it was online. I wonder if the issue with the auto-tagging was born in that piece of the code. If that is the case, my request would be to change it.

There is no explanation or complex coding needed if we can all influence what the front end shows by using the checkboxes. In any case, I cannot implement this latest version until this issue is fixed.

Thanks for your consideration,

Nicky

**Comment by tech2serve** [comment]
> I am entering this as a bug but I realise it might not be considered that by the code-writers.
> 
> In short:
> it seems the newest TSML version auto-tags meetings and this is causing random errors.

If you believe you have experienced an error, please provide as many details as possible so we can investigate it further.

> I would like to know what is triggering the tagging and with that I mean: is a meeting online or in person or both?
> 
> I have a lot of meetings, after updating on my staging site, that show 'in person' and there is nothing in the meeting information that justifies that.
> 
> I would like to get more clarity about what triggers the tagging.

The approach taken is reasonably straightforward. First, let me say that *in person* is not a concept that was supported previously. Prior to COVID, *every* meeting in the database was in person. The plugin did not support other attendance options. And, all entered addresses had to successfully geocode in order for the meeting to publish.  

Over the past year or so we’ve added support for online meetings. The flag indicating the meeting as online was triggered by completion of the Conference URL.  The admin could check a box to flag a meeting location as temporarily closed (not meeting in person), but since early on in the pandemic, *online* has been determined by the data the admin entered.

With COVID waning and most groups opening back up—but with many areas still requiring support for online meetings—we implemented the changes groups asked for to flag meetings as *in person*, *online*, or *hybrid*. 

So what causes a meeting to be flagged as *in person*, *online* or *hybrid*? This decision remains made by the admin through answering the simple question *and* entering the required information:

- For a member to attend a meeting online, that meeting must have a conference URL or conference phone. That has not changed from previous versions. So if the conference URL/phone is *not* present, the meeting should not be flagged as online or it would be misleading to the user.

- For a member to attend a meeting in person, that meeting must have a specific (non approximate address) *and* the admin must have answered in the affirmative that the meeting can be attended in person (behind the scenes, this clears TC but primarily for backward compatibility in the JSON feed).  If the admin says “no” to this question, that is the same as checking TC if a specific address is provided.

- If the admin checks “no” to the question and enters an approximate address, they must include online meeting information in order for the meeting to be valid. When one thinks about this, it makes sense. If I can’t attend a meeting in person, then I need online meeting information in order to attend it.

- But if the admin checks “no” to the question and enters a specific address, that is the same as checking TC; it is how the admin tells the plugin the meeting is temporarily inactive. 

> 
> Ideally I would like all coding to use the check box options to draw conclusions and not do so in a way that is not transparent for me as the admin of the website.

What may not be understood is that validation is important in this process because we feed Meeting Guide (for most sites). For example, we had a number of admins who thought if they simply checked ONL (in the beginning when we handled it this way), the meeting would show up as online. Of course, that doesn’t make sense if online meeting information isn’t provided. 

The same is true for in-person meetings. An admin could check a block for in-person and then expect that flag to show up even if they didn’t provide a specific address. In effect, it is a false label; the user needs a specific address in order to attend the meeting.

> 
> I don't know the coding, but I remember there was a change during Covid at some point, where we no longer had to indicate whether a meeting was online or not. The coding, I was told, just checked if there was a valid URL for Zoom (and similar online platforms) to come to the conclusion that it was online. I wonder if the issue with the auto-tagging was born in that piece of the code. If that is the case, my request would be to change it.
> 
> There is no explanation or complex coding needed if we can all influence what the front end shows by using the checkboxes. In any case, I cannot implement this latest version until this issue is fixed.

If there are problems with the data not getting flagged as we intend, we’ll fix that. At this time we don’t have confirmation of data issues. 

> 
> Thanks for your consideration,
> 
> Nicky

I’m going to close this but will gladly reopen it if we have specific meeting data that doesn’t appear to conform to the logic I’ve outlined above. As I said, please provide as many specifics of the particular meeting you think isn’t being flagged correctly and the data you’ve entered.

Thank you for giving the new version a whirl. I do believe, unless we have a bug (which is possible) that this version provides more options for admins while improving data validation, and gives users asked-for interface changes for finding meetings with the attendance options they wish to use.

Tim

**Comment by brianw-area24** [comment]
@tech2serve gave a good explanation here. I'd like to add a little more that may help. Basically, we look at three different items to determine how a person could attend, and each of those have 2 options each. Therefore, there are only 8 different combinations (2x2x2 = 8) The attached image shows a table of the different combinations, and the attendance option for each one. If you're not getting these results, then we need to know about that.

![Screen Shot 2021-07-01 at 7 44 39 PM](https://user-images.githubusercontent.com/55821195/124204525-db14d780-daa4-11eb-8404-15c8c0083115.png)

**Comment by brianw-area24** [comment]
Going to try 1 more time. 

@tech2serve gave a good explanation. I'm going to add one more thing that might help. We look a three items to determine how a person could attend, and each of those items have 2 options each. Therefore, there are a total of 8 combinations (2x2x2 = 8) This table shows the different combinations, and the attendance option for each of them.

![Screen Shot 2021-07-01 at 7 44 39 PM](https://user-images.githubusercontent.com/55821195/124205149-40b59380-daa6-11eb-829f-d6f077dcd38c.png)

Brian W

**Comment by OSAwebsite** [comment]
Please do not close this issue yet.

Thank you both, Brian and Tim, for providing more clarity on how tags are determined.
I will get you more info tomorrow about the meetings that show up incorrectly.

One remark/observation in advance:
In the new version, the checkbox for Temporarily Closed has disappeared.
And a new question (instead of a checkbox) has been added to the backend, to see if a meeting is in person, online or both.
This forces me (and others) to go through every single meeting post manually to answer that question. I have over 300 meetings to administer and the Wordpress interface is not exactly fast. This is not conducive to quick maintenance and it deviates from the earlier approach to use checkboxes.

If you do not want that, at the very least the question about in-person, online and hybrid should really be included in the import template. Ideally however, I would prefer for you to change the code to include this in the checkboxes, as this would keep the principles of for admins the same.

I completely understand that as free user of this plugin we need to count our blessings every single day for having access to it. However, it seems to me that it might take some value away from the plugin if the development choices are determined in large part by the feeding of data to the meeting guide.

If this is the priority for the future, then I wonder if it is possible to do this the other way around:
Give admins checkboxes for choices and, specifically for feeding data to the meeting guide, translate those choices in the yes/no answers that are now introduced as a new feature.

I understand that no development of any new software version is ever done without discussion everything extensively beforehand. But, as a user that, for maintenance, depends on those choices, I think the user friendliness of the plugin would benefit a lot from choosing one method for admins to make their choices.

I hope you will keep this topic open so that I can actually have a closer look at this over the weekend and provide you with details. But I think that the fact that the disappearance of the checkbox for Temporarily Closed is at the root of the issue here.

I am a software developer myself and I have fallen in this trap myself as well: thinking that automating certain conclusions based on user data was a good thing. And then to find out that users do not want software to draw conclusions unless they can directly influence them. The table that Brian posted shows that the logic is too complicated. Give the control to the admins of the websites. That is what their job is.

Thanks again for your consideration.

**Comment by OSAwebsite** [comment]
OK, as promised, I looked at my staging site again.

Here is a screenshot of the situation of a listing after upgrading the plugin:

[](url)
![Listing_after_upgrading_plugin](https://user-images.githubusercontent.com/59859095/124297891-98491300-db29-11eb-9f80-b4f0aab976b4.jpg)
[](url)

The tagging is wrong. There is nothing in the meeting details that justifies the tag "in person".

After going into the backend of this meeting, I see that the Temporarily Closed checkbox has gone.
I see the new yes/no option to indicate whether the meeting is in person or online.
I reset that option to no and this changes the listing to the situation in the screenshot below:

![Listing_after_correcting_yesno](https://user-images.githubusercontent.com/59859095/124298174-f118ab80-db29-11eb-9c2c-6d64f45d10fa.jpg)

The meeting, by default, has a meeting address. It is strike through on the meeting details page and no map is shown.
If I answer 'yes' to the new question and have a valid URL for Zoom or similar online platforms, the map comes back.

That makes sense.

So now what remains is my request to please please please include the 'in person' question as an import field so that I can bulk adjust my files and refresh the whole database by importing my 300 plus meetings. This saves me a ton of work.

I am not sure that this post is being read by the dev team because you closed it, so I will make this a new topic with the same notes as this post. Sorry for the repetition.

Without the option to make the new setting part of the import, I will not upgrade to the newest version because of the amount of repetitive admin work to be done, especially because the starting defaults after upgrade are randomly incorrect and I do not find that acceptable, especially not without the option to re-import.

Thanks for your consideration.

**Comment by brianw-area24** [comment]
Don't create a new issue, we're still reading this.

Those last images you sent were extremely helpful! However, I'd ask for at least one more image. On your existing, live site, can you edit that (elec) Faith Hope & Serenity AFG meeting, and take a screen shot of the edit screen? I'm especially interested in seeing the "types" section.

I think I might know what's going on, but I want a little more information.

**Comment by OSAwebsite** [comment]
Sorry, Brian, I just created a new issue that actually might contain info
that you find useful.
Adding to that I am attaching a third screenshot of old and new when using
the different combination and making intentional errors in the settings, to
show how this implicit coding can cause unexpected things.
When I entered a non specific address for an online meeting, the address
was interpreted in a generic way showing the geographical location, but
then the tag 'temporarily inactive' was shown. And this is for the same
valid online meeting that is an example of a very realistic scenario: meets
online, but credentials not published.

Two more attached screenshots from the live meeting's backend, showing the
check boxes and the address section. Note that I use the generic Zoom.us
url to force, in the current version, the 'online' tag to be applied.
This is a meeting that does not publish their Zoom details - people have to
phone or email to get them.

Hope this helps - good luck with this and thank you for looking into this.


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Comment by brianw-area24** [comment]
Hey Nicky, I'm not seeing the two screen shots from the backend. Can you upload them again.
Thanks
Brian W

**Comment by tech2serve** [comment]
> OK, as promised, I looked at my staging site again.
> 
> Here is a screenshot of the situation of a listing after upgrading the plugin:
> 
> 
> ![Listing_after_upgrading_plugin](https://user-images.githubusercontent.com/59859095/124297891-98491300-db29-11eb-9f80-b4f0aab976b4.jpg)
> 
> The tagging is wrong. There is nothing in the meeting details that justifies the tag "in person".
>

The fact that a meeting has a specific address associated with it means it is potentially available in person.  This will be true as long as the TC type is not included. 

> After going into the backend of this meeting, I see that the Temporarily Closed checkbox has gone.

Looking at your screen caps, I don’t see anything indicating TC was checked.  As Brian requested, please provide a screen cap of the backend for that meeting showing the types. I suspect it isn’t present on your live site.

I see two options for you:

1. If this meeting in question, and others like it, are natively online meetings, change the address to an approximate address. Then when you upgrade to 3.11, the meeting shouldn’t be interpreted as online.
2. If the meeting still have a physical in person location assigned to them that is actually temporarily not available, add the TC type to those meetings before upgrading to 3.11. Those meetings will then show as Location Temporarily closed in 3.10, but will flagged as Temporarily Inactive (online) when migrated.

**Comment by brianw-area24** [comment]
Looking at the images of the (elec) Faith, Hope, & Serenity AFG - Orillia (D1) meeting, Sunday 8:00pm, the new version of the Plugin is doing exactly as what I would expect it to do. I say this, because on your live site, this meeting has a Zoom Url, a specific address, and it does not have the "Location Temporarily Closed"  flag. Because of the Zoom Url, it meets online, and because of the specific address and not being temporarily Closed, it meets in person. Therefore it's coming into the new version of the plugin as "In-person and Online", exactly as expected.

You've asked that we include the new question in the import. We're not going to do that at this time, because we use the Location Temporarily Closed tag for this. Adding another field to the import would be redundant. In 3.11.x, we're directly storing your answer to that question, instead, we're using that question to help determine whether to include the "Location Temporarily Closed" flag. 

I'm pretty sure we can accomplish this with the current version. Here's what I'd like you to do:
- On your live site, go to the Settings and Imports page, and export a CSV file
- Edit the CSV file, and for any meeting that's Online ONLY, add "Location Temporarily Closed" to the flags if it's not already there
- Import this modified CSV file into your staging server.
I'm thinking if you do this, all these meetings will show up on the staging server as Online only as desired.

Looking forward to hearing your results of this test.

Brian W

**Comment by OSAwebsite** [comment]
I will do that tomorrow, Brian and let you know what the results are.
Thanks for following up!


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Comment by OSAwebsite** [comment]
That's a good catch, Brian, about the missing checkbox for Location
Temporarily Closed in that meeting.
I do wonder though how transparent it is to not show it on the backend, but
still use it when it is present in the import.
That is the thing I was talking about when I used the word implicit versus
explicit coding.
If this is a tag that has significant influence on the front end and as
such it would make things a lot easier to follow if it would remain visible
in the checkboxes.
From what I can see it has disappeared. So, if you make a mistake like
mine, how would I even be able to correct it in the new version's back end?

Anyway, will go through my meetings with a finecomb and see if all of those
with incorrect tag have that same issue. I never thought to look at it like
that.

Talk later!


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Comment by OSAwebsite** [comment]
Today got away from me, Brian, so sorry. Will get back to you tomorrow!


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Comment by OSAwebsite** [comment]
Here are my findings. My main discovery is that the 'in person' tag is
never shown on the list. I would like to see it there.
I have found that the requirements for listings before the upgrade is done
catch a lot of omissions and errors in the data, but due to the
somewhat implicit logic in the code of the new version, it took me a good
number of hours to get my head around it. With 300 plus meetings that was a
lot of work. I did hold on to my own naming convention (online, re-opened,
tc for temporarily closed and opening status unknown) because that just
makes things easier and more transparent for me.

Let me know what you think.

Sorry for the delay...


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Comment by OSAwebsite** [comment]
The only other issues that remain:
- navigating back to the list reverts back to default and not to the
selection last made.
- the filterboxes, especially the open search box. It seems to be wonky or
else maybe I need some information on that.
Example:
When I look for region Toronto, I also find Etobicoke. I realise that both
really are the same general area, but it is not what I am after when I
enter Toronto in my open search box.
Then, when I tried to enter 'near location' and entered Toronto, within 2
km, it gave me just Toronto. Then when, with "Toronto ON" in the search box
I then removed the 2 km and chose 'search' from the search box, I got no
results. In both cases I did not enter a specific day.

This may be splitting hairs, but what I would expect is this:
With nothing in the search box, I either look for a district or a region or
a specific day, time and meeting type.
With criteria in the search box, selecting 'search', without selecting
either a region or a specific group, I expect to find a wider result.
When I search for the region of Toronto, I expect to find meetings with a
meeting address in Toronto (not Etobicoke).
It is not so important if these expectations are correct, but it would be
helpful to know what the code-assumptions are, so I can explain this to
users.

Thanks and good luck with the development!






-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Comment by tech2serve** [comment]
> Here are my findings. My main discovery is that the 'in person' tag is never shown on the list. I would like to see it there. I have found that the requirements for listings before the upgrade is done catch a lot of omissions and errors 

What list? It shows up on the public meeting list users see. If you’re not seeing it, please [open a new Discussion](https://github.com/code4recovery/12-step-meeting-list/discussions):

![E047D86C-861F-427D-8981-19EE9A62B52F](https://user-images.githubusercontent.com/31904316/124529088-977cdf00-ddbe-11eb-9dcd-855de5261f8f.jpeg)

NOTE: We’re implementing the new GitHub feature _Discussions_ as a way to separate concerns and provide more community discussion.

> The only other issues that remain: - navigating back to the list reverts back to default and not to the selection last made. - the filterboxes, especially the open search box. It seems to be wonky or else maybe I need some information on that. Example: When I look for region Toronto, I also find Etobicoke. I realise that both really are the same general area, but it is not what I am after when I enter Toronto in my open search box. Then, when I tried to enter 'near location' and entered Toronto, within 2 km, it gave me just Toronto. Then when, with "Toronto ON" in the search box I then removed the 2 km and chose 'search' from the search box, I got no results. In both cases I did not enter a specific day. This may be splitting hairs, but what I would expect is this: With nothing in the search box, I either look for a district or a region or a specific day, time and meeting type. With criteria in the search box, selecting 'search', without selecting either a region or a specific group, I expect to find a wider result. When I search for the region of Toronto, I expect to find meetings with a meeting address in Toronto (not Etobicoke). It is not so important if these expectations are correct, but it would be helpful to know what the code-assumptions are, so I can explain this to users. Thanks and good luck with the development! -

Nothing really changed in this area of the code except to add the ability to filter by _in person_, _online_ or _active_, and we are aware of a bug here (See #396).

**Comment by OSAwebsite** [comment]
That's a good idea, the 'discussion' category!


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Comment by OSAwebsite** [comment]
For now keeping it in this subject, just in case I am missing something:
here are 2 screenshots.
Showing the list of meetings I have set and know to be re-opened (in
person, or in person and online).
As you can see, only the hybrids show up in green. The others are supposed
to show in person, as far as I am concerned.
The second screenshot shows the details page of such a meeting and as you
can see the tag in person IS in fact shown there.
So I would like to see it on the list as well.

I am still using open status references in the meeting names, so I am sure
that all of these have the required settings to produce the tag 'in person'.


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Comment by brianw-area24** [comment]
I'm not seeing the screenshots, but I think I know what you're requesting, which is that it show "In-person" on the screen that lists all the meetings. Our assumption is that every meeting is in-person, unless told otherwise, therefore it would be be redundant to say "In-person".

I think part of it depends on what percent of your meeting are meeting in person. For a site like yours, where most meetings are not yet, an "In-person" would probably be OK. However, for a site like the one I work on, where most of the meetings are back to meeting in person, it would get very cluttered to say "In-person" over and over again. See the [Central Iowa AA Page](https://www.aacentraliowa.org/meetings?tsml-day=any)

Over time, and hopefully quickly, more and more meetings will be going back to meeting in person, so this should become less and less of an issue.

Brian W

**Comment by OSAwebsite** [comment]
Hi Brian,

Sorry to hear you were unable to see the screenshot but yes, you understood
correctly.
This issue is probably my main beef with the plugin although I am at the
same time counting my blessings that it has been made available outside AA.
The beef being that I feel a tool should be consistent in what it does and
not change just because within AA most meeting are back to in-person. It is
sort of subjective programming, while, at the same time, the TSML web
development has made a promise in the past to have this always available to
all 12 step programs. Not all programs are equal... Again: I realise that
the origin of the tool lies in AA and as such far be it for me to complain.
It would be an absolute nightmare if I had to find or even create my own
alternative. It would never be to the standard of the plugin.

Having said this, I simply want to draw your attention to my train of
thoughts:
With tags for meeting, I think that, while we are still so much affected by
Covid, we stick to showing this tag on the list. It is there in a small
front, big enough to notice, and I do not interpret that as clutter. That's
just my view on it. The other reason to want it is that if you mention
in-person and online explicitly it just makes me wonder why the others
don't have a tag. For me, that is the sole reason I have adjusted all
meeting names with my own tag and I do think my visitors use that more than
the plugin tag.

So, in short: I understand what you are saying but I would like the team to
always keep in mind that it is no longer just programming for AA...

Thank you for  your consideration, Brian! Very much appreciated!

Nicky


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*
